spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -112,6 +112,246 @@ Subsequently, an implementation of Jakarta Data, specifically tailored to the ch
 
 Jakarta Data empowers developers to shape their data access strategies by defining entity classes and repositories, with implementations seamlessly adapting to the chosen data store. This flexibility and Jakarta Data's persistence-agnostic approach promote robust data management within Java applications.
 
+=== Repository interfaces
+
+A Jakarta Data repository is a Java interface annotated `@Repository`.
+A repository interface may declare:
+
+- _abstract_ (non-`default`) methods, and
+- _concrete_ (`default`) methods.
+
+A concrete method may call other methods of the repository, including abstract methods.
+
+Every abstract method of the interface is usually either:
+
+- an entity instance _lifecycle method_,
+- an _annotated query method_,
+- an _automatic query method_, or
+- a _resource accessor method_.
+
+A repository may declare lifecycle methods for a single entity type, or for multiple related entity types.
+Similarly, a repository might have query methods which return different entity types.
+
+A repository interface may inherit methods from a superinterface.
+A Jakarta Data implementation must treat inherited abstract methods as if they were directly declared by the repository interface.
+For example, a repository interface may inherit the `CrudRepository` interface defined by this specification.
+
+[NOTE]
+A Jakarta Data provider might go beyond what is required by this specification and support abstract methods which do not fall into any of the above categories.
+Such functionality is not defined by this specification, and so programs with repositories which declare such methods are not portable between providers.
+
+The subsections below specify the rules that an abstract method declaration must observe so that the Jakarta Data implementation is able to provide an implementation of the abstract method.
+
+- If every abstract method of a repository complies with the rules specified below, then the Jakarta Data implementation must provide an implementation of the repository.
+- Otherwise, if a repository declares an abstract method which does not comply with the rules specified below, or makes use of functionality which is not supported by the Jakarta Data implementation, then an error might be produced by the Jakarta Data implementation at build time or at runtime.
```

Comment:
Component specs can have optional features. It is the platform and profiles that cannot have optional features, and so have to either explicitly state that such optional features from a component spec is either not required and excluded from the platform/profile, or to state that it is required behavior.

Reply:
> Component specs can have optional features. It is the platform and profiles that cannot have optional features, and so have to either explicitly state that such optional features from a component spec is either not required and excluded from the platform/profile, or to state that it is required behavior.

That means we could technically include this as an optional feature, but it will burden the Jakarta EE Platform and Web Profile with having to call it out as not required and excluded.  For something this minor, I'd say it's definitely not worth it to include this as an optional feature.  It's only restating what vendors with the optional capability would already be assumed to be doing.

Reply:
@njr-11 I'm not sure I understand. What precisely is the optional feature you mean?

Reply:
> Beyond the optional features, we have extensibility. I can create a Repository specialization that might work with a provider or a specific database.

Yes, it's more this extensibility that I'm trying to describe here. I'm not trying to actually _define_ any optional features, just kind of lay out a framework for how Jakarta Data can be extended by datastore-specific "extensions".

Reply:
This is my mistake. I had misread the section and thought it was referring to something else.  When I re-read it now, it looks great.  There is no reason to take this out.  It's only calling out that Jakarta Data providers might choose to extend the spec by allowing additional abstract methods, which is fine.  Please disregard my prior comment on this one. I'll mark the conversion resolved so it doesn't hold up the pull.